### PR TITLE
 Procedural Map Generator Improvements - tgstation/-tg-station#9252

### PR DESCRIPTION
**Pull request by tigercat2000**
>- Fixes a bug where turf cluster checks checked for atoms instead of turfs
- Changes to the clusterCheckFlag system by replacing + adding new defines
- Adds a proc `defineCircularRegion()` to define a circular map region, CIRCLES and SPHERES!
- `defineRegion()` and `defineCircularRegion()` no longer replace the map eniterely, they now add all the new turfs to the map list (no duplicates), the ability to overwrite the map has been moved to a new argument on both procs replace
- Adds a proc `undefineRegion()` which just makes the map list a new list, it's just there to look nice
- Removes bottomLeft and topRight, they weren't even needed
- Made the turf clustering checks faster by remembering that `trange()` exists, and actually using it.

So TL:DR; Circle mapgen, bugfixes, faster stuff, undefining regions.

tgstation/-tg-station#9252

### CODE REVIEW

**Comment by Spacemanspark** [comment]
:+1:

**Comment by tigercat2000** [comment]
for funsies, will post a screenshot of what I have been using the mapgen for
xenocottage 2

**Comment by fj45** [comment]
We really need to use that.

**Comment by RemieRichards** [comment]
@tigercat2000 wow wow, what in that picture exactly is generated?

**Comment by tigercat2000** [comment]
@RemieRichards Used the nature gen for a big space, turned up a mob's luminosity to 10 and ran around to get the unsimmed grass to load lighting, used build mode to make a clearing and the walls, used build-mode's edit-mode to make the unsimmed grass's icon_state as wood flooring, and placed the lamps myself.
So, most of it is build-mode shenanigans.

Nature gen was pretty useful for getting the forest laid out though, couldn't have done it without it.

**Comment by RemieRichards** [comment]
@tigercat2000 
Neato, I'm looking at my formulas for spherical ranges and I must have been on crack when I wrote them because they look to be dodgy, so I'll be updating my PR should tests prove me right.

**Comment by RemieRichards** [comment]
@tigercat2000 yep, you'll need the latest commit, it fixes spheres being off-center and corrects illogical spheres. the fixes was that centerZ was nowhere close to the center, and was actually some dark magic value connected to each constructed sphere. it's magic.
